sympy/solvers/tests/test_solveset.py

```diff
@@ -1058,6 +1058,8 @@ def test_issue_9953():
     assert linsolve([ ], x) == S.EmptySet
 
 
+def test_issue_10534():
+    assert solveset_real(Piecewise((x, y<0), (x + 1, True)), x) == FiniteSet(-1,0)
```

Comment:
I don't think we can give a FiniteSet solution to Piecewise. -1, for example, is not always a solution. It's only a solution when `y < 0`.

Reply:
In the example you cite the conditions are expressed in terms of  x whereas here they are in y. The y conditions create different functions to be solved. Substituting a y into the initial eq and solving should give a set the same as when the order of operation is reversed (solve and feplace).